applications/main/bad_usb/bad_usb_script.c

```diff
@@ -205,16 +208,38 @@ static bool ducky_altstring(const char* param) {
     return state;
 }
 
-static bool ducky_string(const char* param) {
+static bool ducky_string(BadUsbScript* bad_usb, const char* param) {
     uint32_t i = 0;
-    while(param[i] != '\0') {
-        uint16_t keycode = HID_ASCII_TO_KEY(param[i]);
-        if(keycode != HID_KEYBOARD_NONE) {
-            furi_hal_hid_kb_press(keycode);
-            furi_hal_hid_kb_release(keycode);
+    uint32_t timing = 0;
+
+    if(bad_usb->stringdelay == 0) {
+        timing = 0;
+        while(param[i] != '\0') {
+            uint16_t keycode = HID_ASCII_TO_KEY(param[i]);
+            if(keycode != HID_KEYBOARD_NONE) {
+                furi_hal_hid_kb_press(keycode);
+                furi_hal_hid_kb_release(keycode);
+            }
+            i++;
+        }
+    } else if(bad_usb->stringdelay > 0) {
+        if(bad_usb->stringdelay > 1000) {
+            timing = bad_usb->stringdelay / 2 / 10;
```

Comment:
This code doesn't looks good: when we designing something we expect it to be predictable.
In this case function changes its behavior when param exceeds some value.
And to be honest I don't see any reason why it should.